app/components/model-renderer.jsx

```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import Model from './modelling';
+
+// function to check whether webgl is available
+function webGLCompatibilityTest() {
+  try {
+    return !!window.WebGLRenderingContext && !!document.createElement('canvas').getContext('experimental-webgl');
+  } catch (e) {
+    return false;
+  }
+}
+
+class ModelRenderer extends React.Component {
+  constructor(props) {
+    super(props);
+    this.getDifference = this.getDifference.bind(this);
+    this.setTexture = this.setTexture.bind(this);
+
+    this.imgUrl = Object.values(this.props.subject.locations[0])[0];
```

Comment:
Why is this setting an instance property instead of component state?

Reply:
I don't think `this.props.subject.locations[0])[0]` is guaranteed to be the image you want, or even an image, for subjects in general.